fix: Add device properties addition and notify removal (#1160)
edgexfoundry/edgex-docs: docs_src/microservices/device/V3Migration.md

3. ProtocolProperties now supports typed values.

    !!! example "ProtocolProperty with typed values"

        ```yaml
        protocols:
          other:
            Address: simple01
            Port: 300
        ```

> 4. The boolean field `notify` is removed as device creation notification is handled by system event message now.

Reviewer: ```suggestion
4. The boolean field `notify` is removed as it is never used.
```

3. ProtocolProperties now supports typed values.

    !!! example "ProtocolProperty with typed values"

        ```yaml
        protocols:
          other:
            Address: simple01
            Port: 300
        ```

4. The boolean field `notify` has been removed as it is never used.
5. An extendable field `properties` has been added to Device.

> 6. Added `tags` field from event level to Device.

Reviewer: ```suggestion
6. Added `tags` field to Device for event level tagging.
```

### Device Profile Files

1. Add `optional` field in ResourceProperties to allow any additional or customized data.
2. Change the data type of `mask`, `shift`, `scale`, `base`, `offset`, `maximum` and `minimum` from string to number in ResourceProperties.

> 3. Added `tags` field from reading level in DeviceResource

Reviewer: ```suggestion
3. Added `tags` field in DeviceResource for reading level tagging.
```

> 4. Added `tags` field from event level in DeviceCommand

Reviewer: ```suggestion
4. Added `tags` field in DeviceCommand for event level tagging.
```

            testPropertyB: meter
        ```

> 6. ProvisionWatcher can be added an extendable field `Properties map[string]`.

Reviewer: ```suggestion
6. ProvisionWatcher is added an extendable field `Properties`.
```

Author: udpated

            testPropertyB: meter
        ```

> 6. An extendable field `properties` has been added to ProvisionWatcher.

Author: updated

    ```go
    import (
    	...

>         

Reviewer 2: This indent is require for rendering properly

            5.67

            or

>             

Reviewer 2: Please restore all these removed indentation which are required.
